### PR TITLE
GH-250 Implement Launcher API (#356)

### PR DESCRIPTION
**Pull request by jamesmontemagno**
### Description of Change ###

Describe your changes here. 

### Bugs Fixed ###

- Related to issue #356

Provide links to issues here. Ensure that a GitHub issue was created for your feature or bug fix before sending PR.

### API Changes ###

List all API changes here (or just put None), example:

Added: 
 
- string - string Class.Property { get; set; } 
- void Class.Method();

Changed:

 - object Cell.OldPropertyName => object Cell.NewPropertyName

### Behavioral Changes ###

Describe any non-bug related behavioral changes that may change how users app behaves when upgrading to this version of the codebase.

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

**Comment by dend** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[DeviceTests/DeviceTests.Shared/DeviceTests.Shared.projitems](https://github.com/xamarin/Essentials/blob/feature/launcher/DeviceTests/DeviceTests.Shared/DeviceTests.Shared.projitems) | :white_check_mark:Succeeded | |
[DeviceTests/DeviceTests.Shared/Launcher_Tests.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/DeviceTests/DeviceTests.Shared/Launcher_Tests.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/View/LauncherPage.xaml](https://github.com/xamarin/Essentials/blob/feature/launcher/Samples/Samples/View/LauncherPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/View/LauncherPage.xaml.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Samples/Samples/View/LauncherPage.xaml.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/HomeViewModel.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Samples/Samples/ViewModel/HomeViewModel.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/LauncherViewModel.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Samples/Samples/ViewModel/LauncherViewModel.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.android.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.ios.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.netstandard.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.netstandard.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.shared.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.uwp.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.uwp.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/Launcher.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/Xamarin.Essentials/Launcher.xml) | :white_check_mark:Succeeded | |
[docs/en/index.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/index.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C23%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201807232021366701-405%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=QeBUUlPOYe2xV%2BiJm7XPp2AmZgB0gHkeJOPQUNZEkQg%3D&st=2018-07-23T20%3A17%3A52Z&se=2018-08-23T20%3A22%3A52Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: DeviceTests/DeviceTests.Shared/Launcher_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+using Xunit;
+
+namespace DeviceTests
+{
+    // TEST NOTES:
+    //   - a human needs to close the browser window
+    public class Launcher_Tests
+    {
+        [Theory]
+        [InlineData("http://www.example.com")]
```
Comment (Redth) [inline DeviceTests/DeviceTests.Shared/Launcher_Tests.cs]: Some thoughts on more inline data tests:

```
https://www.example.com
http://example.com/?query=blah
https://example.com/?query=blah
mailto://someone@microsoft.com
mailto://someone@microsoft.com?subject=test
```

Comment (jamesmontemagno) [reply DeviceTests/DeviceTests.Shared/Launcher_Tests.cs]: Should be good now.

**Comment by dend** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[DeviceTests/DeviceTests.Shared/DeviceTests.Shared.projitems](https://github.com/xamarin/Essentials/blob/feature/launcher/DeviceTests/DeviceTests.Shared/DeviceTests.Shared.projitems) | :white_check_mark:Succeeded | |
[DeviceTests/DeviceTests.Shared/Launcher_Tests.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/DeviceTests/DeviceTests.Shared/Launcher_Tests.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/View/LauncherPage.xaml](https://github.com/xamarin/Essentials/blob/feature/launcher/Samples/Samples/View/LauncherPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/View/LauncherPage.xaml.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Samples/Samples/View/LauncherPage.xaml.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/HomeViewModel.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Samples/Samples/ViewModel/HomeViewModel.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/LauncherViewModel.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Samples/Samples/ViewModel/LauncherViewModel.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Browser/Browser.android.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Browser/Browser.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Browser/Browser.ios.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Browser/Browser.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Browser/Browser.uwp.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Browser/Browser.uwp.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.android.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.ios.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.netstandard.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.netstandard.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.shared.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.uwp.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.uwp.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/Launcher.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/Xamarin.Essentials/Launcher.xml) | :white_check_mark:Succeeded | |
[docs/en/index.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/index.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C25%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201807251440529442-405%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=ap4yD5CG2V5OE43zDdH8yIDIe7a6hp2Dm0d0NW6%2BNgI%3D&st=2018-07-25T14%3A37%3A05Z&se=2018-08-25T14%3A42%3A05Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by dend** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[DeviceTests/DeviceTests.Shared/DeviceTests.Shared.projitems](https://github.com/xamarin/Essentials/blob/feature/launcher/DeviceTests/DeviceTests.Shared/DeviceTests.Shared.projitems) | :white_check_mark:Succeeded | |
[DeviceTests/DeviceTests.Shared/Launcher_Tests.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/DeviceTests/DeviceTests.Shared/Launcher_Tests.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/View/LauncherPage.xaml](https://github.com/xamarin/Essentials/blob/feature/launcher/Samples/Samples/View/LauncherPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/View/LauncherPage.xaml.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Samples/Samples/View/LauncherPage.xaml.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/HomeViewModel.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Samples/Samples/ViewModel/HomeViewModel.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/LauncherViewModel.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Samples/Samples/ViewModel/LauncherViewModel.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Browser/Browser.android.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Browser/Browser.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Browser/Browser.ios.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Browser/Browser.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Browser/Browser.uwp.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Browser/Browser.uwp.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.android.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.ios.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.netstandard.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.netstandard.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.shared.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Launcher/Launcher.uwp.cs](https://github.com/xamarin/Essentials/blob/feature/launcher/Xamarin.Essentials/Launcher/Launcher.uwp.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/Launcher.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/Xamarin.Essentials/Launcher.xml) | :white_check_mark:Succeeded | |
[docs/en/index.xml](https://github.com/xamarin/Essentials/blob/feature/launcher/docs/en/index.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C25%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201807251453032180-405%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=kWZsAkxPCZ6njWtw795vBO8LMUS08LIh9aBVkoRSsMc%3D&st=2018-07-25T14%3A49%3A13Z&se=2018-08-25T14%3A54%3A13Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**